pkg/types/media.go

```diff
@@ -17,6 +17,7 @@ package types
 
 const (
 	CycloneDXMediaType     = "application/vnd.cyclonedx"
+	SyftMediaType          = "application/vnd.syft"
```

Comment:
Actually for this we might need to check with @luhring - should this be something like `application/vnd.com.anchore.syft`?

Reply:
Good question — we just recently registered `application/vnd.syft+json` with the IANA (https://www.iana.org/assignments/media-types/application/vnd.syft+json)

Reply:
Perfect! @samj1912 could you add the +json part and then we should be good to merge!

Reply:
Sure, I didn't add it since we don't add the file extensions suffixes for cyclonedx or spdx either (spdx recently added application/spdx+json for their json documents). I was hoping to make a separate PR to add file extension suffixes for all of them. But can do it for syft specifically in this PR.

Reply:
Related https://github.com/spdx/spdx-spec/issues/577 and https://github.com/anchore/syft/issues/612

Reply:
Ah ok. I'm not an expert here. There was actually some discussion about whether or not we were supposed to add it or not for the others and I can't remember the explanation.

Reply:
@dlorenc done - should be good to go. Will also try to update the others in another PR.

Reply:
>There was actually some discussion about whether or not we were supposed to add it or not for the others

If we do find this, I'd be interested in learning more! I'm also not an expert on this.

From my understanding, it's important to have the `+json` part. This is because without it 1) it's no longer a registered media type, and 2) content interpretation becomes ambiguous to consumers of the data.

And I'd guess this applies equally to the already listed media types (e.g. `application/vnd.cyclonedx`) just as well, so I like your idea of a follow-up PR to address this, @samj1912!

Reply:
Found it! It's long and confusing: https://github.com/sigstore/cosign/pull/439#discussion_r671834751

My understanding is that the suffixes are probably correct to add, but only if they were actually registered? Unless you're in the vendor sub-tree, where it might be ok, depending on which subtype you're using.

Reply:
That's a great tl;dr. Thanks for the link!

Is this good to merge at this point?

Reply:
Yep - sorry. There was a test flake I had to rerun.